### DISCUSSION TITLE
How different Entities use different shader variables

### DISCUSSION THREAD

**Discussion by C0j23T** [bevyengine/bevy#14754] [Q&A]
If I have 500 entities with a specific component, is it possible to provide different shader uniform variables for the 500 entities based on the value of the component? For example, let 500 entities use different colors.

**Comment by ncthbrt** [comment]
You can see this for an example: https://webgpu.github.io/webgpu-samples/?sample=instancedCube#instanced.vert.wgsl

**Reply by C0j23T** [reply]
Um...For example I have this component:
``` rust
#[derive(Component)]
struct Color3f(f32, f32, f32);
```
Then I spawned hundreds of cubes with different Color3f components, how can I make the color of the cube to be the same as Color3f, and after Color3f changes the cube will change color as well?
